src/librustc/ty/query/job.rs

```diff
@@ -368,16 +370,9 @@ fn remove_cycle<'tcx>(
         // Reverse the stack so earlier entries require later entries
         stack.reverse();
 
-        // Extract the spans and queries into separate arrays
-        let mut spans: Vec<_> = stack.iter().map(|e| e.0).collect();
-        let queries = stack.into_iter().map(|e| e.1);
-
+        // The stack is a vector of pairs of spans and queries
         // Shift the spans so that queries are matched with the span for their waitee
-        let last = spans.pop().unwrap();
-        spans.insert(0, last);
-
-        // Zip them back together
-        let mut stack: Vec<_> = spans.into_iter().zip(queries).collect();
+        stack.0.swap(0, stack.len() - 1);
```

Comment:
Whoops, I should have written `stack.0.shift_right(1)` here; I'll fix it as soon as I get some CI results (in case of any other issues).